plugin/src/main/java/org/autorefactor/refactoring/rules/SwitchRefactoring.java

```diff
@@ -385,4 +428,96 @@ private Variable extractVariableWithConstantValue(Expression firstOp, Expression
         }
         return null;
     }
+
+    @Override
+    public boolean visit(final SwitchStatement node) {
+        final List<SwitchCaseSection> switchStructure = getSwitchStructure(node);
+
+        for (int referenceIndex = 0; referenceIndex < switchStructure.size() - 1; referenceIndex++) {
+            for (int comparedIndex = referenceIndex + 1; comparedIndex < switchStructure.size(); comparedIndex++) {
+                final SwitchCaseSection referenceCase = switchStructure.get(referenceIndex);
+                final SwitchCaseSection comparedCase = switchStructure.get(comparedIndex);
+
+                if (!referenceCase.stmts.isEmpty()
+                        && breaksControlFlow(referenceCase.stmts.get(referenceCase.stmts.size() - 1))
```

Comment:
(Do not act on this comment, I am fixing it)

```java
                if (!referenceCase.stmts.isEmpty()
                        && breaksControlFlow(referenceCase.stmts.get(referenceCase.stmts.size() - 1))
```
This check is independent of the `comparedCase`, so it should be moved outside of the inner for loop.